ui/app/src/model/datasource-api.ts

```diff
@@ -216,7 +216,7 @@ export class CachedDatasourceAPI implements DatasourceApi {
 }
 
 // Helper function for getting a proxy URL from a datasource or global datasource
-function getProxyUrl(datasource: Datasource | GlobalDatasource) {
+function getProxyUrl(datasource: ProjectDatasource | GlobalDatasource) {
```

Comment:
You should reuse the `Datasource` type you created here

`export type Datasource = ProjectDatasource | GlobalDatasource;`